[Console] Add onboarding tour

ElenaStoeva opened the pull request:
## Summary

This PR adds an onboarding tour to Console and stores its state in local storage.




https://github.com/user-attachments/assets/80251779-9276-488e-9e72-656b84912a2e

ElenaStoeva committed:
[Console] Add onboarding tour

ElenaStoeva committed:
Refactor changes

ElenaStoeva committed:
Add functional tests

ElenaStoeva committed:
Remove subtitle

ElenaStoeva committed:
Merge branch 'console-phase-2' into console-tour

ElenaStoeva committed:
Add last tour step

elasticmachine commented:
Pinging @elastic/kibana-management (Team:Kibana Management)

ElenaStoeva committed:
Merge branch 'console-phase-2' into console-tour

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

ryankeairns commented on src/plugins/console/public/styles/_app.scss:
```diff
@@ -22,6 +22,24 @@
 }
 
+.tourStepFixedContainer {
+  position: fixed; /* Fixed position ensures it stays in view */
```
Could the same be accomplished with the `repositionOnScroll` prop on EuiTour?

ElenaStoeva replied:
Nice catch! It's actually not needed at all! I initially added these stylings to specify a specific position of the second tour step, but I think its position is good as it is. I removed these new classes.

sabarasaba commented on src/plugins/console/public/application/containers/main/get_tour_steps.tsx:
```diff
@@ -0,0 +1,93 @@
+      }),
+      content: i18n.translate('console.tour.editorStepContent', {
+        defaultMessage:
+          'Enter a request in this input pane, and the response will be shown in the neighboring output pane. For more details, visit our QueryDSL documentation.',
```
Shouldnt `QueryDSL documentation` be a link here? 🤔

ElenaStoeva replied:
Yep, I missed this! Addressed in [f64b854](https://github.com/elastic/kibana/pull/191021/commits/f64b854c35c615a5039b9cc826b74ce0e257c19a).

sabarasaba commented on src/plugins/console/public/application/containers/main/get_tour_steps.tsx:
```diff
@@ -0,0 +1,93 @@
+      }),
+      content: i18n.translate('console.tour.historyStepContent', {
+        defaultMessage:
+          'Enter a request in this input pane, and the response will be shown in the neighboring output pane. For more details, visit our QueryDSL documentation.',
```
I think this copy should be: `The history panel keeps track of your past queries, making it easy to review and rerun them.`

ElenaStoeva replied:
Nice catch! Fixed in [f64b854](https://github.com/elastic/kibana/pull/191021/commits/f64b854c35c615a5039b9cc826b74ce0e257c19a).

sabarasaba commented on src/plugins/console/public/application/containers/main/get_tour_steps.tsx:
```diff
@@ -0,0 +1,93 @@
+      }),
+      content: i18n.translate('console.tour.configStepContent', {
+        defaultMessage:
+          'The history panel keeps track of your past queries, making it easy to review and rerun them. ',
```
I think this copy should be: `Fine-tune your Console’s settings and manage variables to personalize your workflow to suit your work style.`

ElenaStoeva replied:
Fixed in [f64b854](https://github.com/elastic/kibana/pull/191021/commits/f64b854c35c615a5039b9cc826b74ce0e257c19a).

sabarasaba reviewed (commented):
Nice work @ElenaStoeva! Left few small comments and also I've noticed a few things while testing locally:

* Seems when loading the page there is an error with the `TopNavMenu`: `Warning: Each child in a list should have a unique "key" prop.`
* When the tour starts, I think the first message we should position below the element otherwise it blocks most of the page navigation and crumbs:

![Screenshot 2024-08-23 at 15 30 01](https://github.com/user-attachments/assets/250127b8-d6c1-478c-83f9-20008db0cac3)

ElenaStoeva committed:
Fix tests and address feedback

ElenaStoeva committed:
Fix console error and anchor positions

ElenaStoeva commented:
Many thanks for the review @sabarasaba! I addressed your comments and fixed the console error and the anchor positions of the tour steps. Could you please take another look?

sabarasaba reviewed (approved):
🚀

julianrosado reviewed (approved):
Looks good to me! I'd also suggest having a UX writer give it a look through to make sure my engrish is good (if you haven't already). Maybe @leemthompo can give it a look?

leemthompo commented on src/plugins/console/public/application/containers/main/get_tour_steps.tsx:
```diff
@@ -0,0 +1,122 @@
+        <EuiText>
+          <FormattedMessage
+            id="console.tour.editorStepContent"
+            defaultMessage="Enter a request in this input pane, and the response will be shown in the neighboring output pane. For more details, visit {queryDslDocs}."
```
```suggestion
            defaultMessage="Enter a request in this input pane, and view the response in the adjacent output pane. For more details, visit {queryDslDocs}."
```

ElenaStoeva replied:
Added with [e4ef690](https://github.com/elastic/kibana/pull/191021/commits/e4ef690b976d63dacdc9903b04260f7720ac4153).

leemthompo commented on src/plugins/console/public/application/containers/main/get_tour_steps.tsx:
```diff
@@ -0,0 +1,122 @@
+        <EuiText>
+          {i18n.translate('console.tour.configStepContent', {
+            defaultMessage:
+              'Fine-tune your Console’s settings and manage variables to personalize your workflow to suit your work style.',
```
```suggestion
              'Fine-tune your Console’s settings and create variables to personalize your workflow.',
```

ElenaStoeva replied:
Added with [e4ef690](https://github.com/elastic/kibana/pull/191021/commits/e4ef690b976d63dacdc9903b04260f7720ac4153).

leemthompo commented on src/plugins/console/public/application/containers/main/get_tour_steps.tsx:
```diff
@@ -0,0 +1,122 @@
+    {
+      step: 4,
+      title: i18n.translate('console.tour.configStepTitle', {
+        defaultMessage: 'Tailor your toolbox',
```
```suggestion
        defaultMessage: 'Personalize your toolbox',
```

I like the alliteration and the Home Improvement vibes but this is slightly clearer :)

leemthompo replied:
it's a coinflip between `personalize` and `customize` 🪙 :-)

julianrosado replied:
Which is more universal (for translation purposes) @leemthompo ?

leemthompo replied:
Probably `customize` by an eyelash :)

ElenaStoeva replied:
Added with [e4ef690](https://github.com/elastic/kibana/pull/191021/commits/e4ef690b976d63dacdc9903b04260f7720ac4153).

leemthompo commented on src/plugins/console/public/application/containers/main/get_tour_steps.tsx:
```diff
@@ -0,0 +1,122 @@
+        <EuiText>
+          {i18n.translate('console.tour.filesStepContent', {
+            defaultMessage:
+              'Easily export your console input requests to a file, or import those you’ve saved previously.',
```
```suggestion
              'Easily export your Console requests to a file, or import ones you’ve saved previously.',
```

ElenaStoeva replied:
Added with [e4ef690](https://github.com/elastic/kibana/pull/191021/commits/e4ef690b976d63dacdc9903b04260f7720ac4153).

leemthompo reviewed (commented):
@julianrosado just a few tweaks from me :)

leemthompo commented on src/plugins/console/public/application/containers/main/get_tour_steps.tsx:
```diff
@@ -0,0 +1,122 @@
+        <EuiText>
+          {i18n.translate('console.tour.shellStepContent', {
+            defaultMessage:
+              'This is our UI for interacting with Elasticsearch clusters using QueryDSL. Easily run queries, manage settings, and view search results.',
```
technically you can call elasticsearch and Kibana apis so we might want to make that explicit?

ElenaStoeva replied:
Thanks for the suggestions! I added them in [e4ef690](https://github.com/elastic/kibana/pull/191021/commits/e4ef690b976d63dacdc9903b04260f7720ac4153).

ryankeairns reviewed (approved):
Thanks for the change

ElenaStoeva committed:
Update copy, fix type errors, refactor changes

ElenaStoeva committed:
Merge branch 'console-phase-2' into console-tour

ElenaStoeva committed:
Add keys to tab tour steps

ElenaStoeva committed:
Merge branch 'console-tour' of https://github.com/ElenaStoeva/kibana into console-tour

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/notice'

leemthompo commented on src/plugins/console/public/application/containers/main/get_tour_steps.tsx:
```diff
@@ -27,7 +28,7 @@ export function getTourSteps(docLinks) {
           {i18n.translate('console.tour.shellStepContent', {
             defaultMessage:
-              'This is our UI for interacting with Elasticsearch clusters using QueryDSL. Easily run queries, manage settings, and view search results.',
+              'Use the Console UI to call Elasticsearch and Kibana APIs and view their responses. Use Query DSL syntax to search your data, manage settings, and more.',
```
```suggestion
              'Console is an interactive UI for calling Elasticsearch and Kibana APIs and viewing their responses. Use Query DSL syntax to search your data, manage settings, and more.',
```

ElenaStoeva replied:
Done! [f9ab9d0](https://github.com/elastic/kibana/pull/191021/commits/f9ab9d0e9df5ff4500630b62599c33ef66d327ce)

leemthompo reviewed (commented):
Looking at this again, I wanted to tweak this copyedit :)

ElenaStoeva committed:
Update copy and fix tests

ElenaStoeva committed:
Skip failing Ace test - not needed anymore

ElenaStoeva commented:
Merging with two CI failed checks, both of which are not related to this PR and will be fixed in https://github.com/elastic/kibana/pull/190961

- `FTR Configs #103 / console app console output panel should be able to copy the response of a request` - Failing because the selecting/copying responses in output panel doesn't work on `console-phase-2` branch
- Type errors in `src/plugins/console/public/application/containers/editor/monaco/monaco_editor_output_actions_provider.ts`
